pkg/didcomm/protocol/outofband/service.go

```diff
@@ -339,32 +342,40 @@ func extractDIDCommMsgBytes(_ *decorator.Attachment) ([]byte, error) {
 	return nil, nil
 }
 
-func decodeInvitationAndRequest(msg service.DIDCommMsg) (*didexchange.Invitation, *Request, error) {
+func decodeInvitationAndRequest(msg service.DIDCommMsg) (*didexchange.OOBInvitation, *Request, error) {
 	req := &Request{}
 
 	err := msg.Decode(req)
 	if err != nil {
 		return nil, nil, fmt.Errorf("failed to decode out-of-band request message : %w", err)
 	}
 
-	invitation := &didexchange.Invitation{
-		ID:    uuid.New().String(),
-		Label: req.Label,
-		Type:  didexchange.InvitationMsgType,
+	invitation := &didexchange.OOBInvitation{
+		ID:       req.ID,
+		ThreadID: req.ID,
+		Label:    req.Label,
 	}
 
-	for _, svc := range req.Service {
-		did, ok := svc.(string)
-		if ok {
-			invitation.DID = did
-			break
-		}
+	target, err := chooseTarget(req.Service)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to choose a target to perform did-exchange against : %w", err)
 	}
 
-	if invitation.DID == "" {
-		return nil, nil, errors.New("ouf-of-band protocol only supports invitations with public DIDs for now")
-	}
+	invitation.Target = target
 
 	// TODO support explicit invitations : https://github.com/hyperledger/aries-framework-go/issues/1502
 	return invitation, req, nil
 }
+
+func chooseTarget(svcs []interface{}) (interface{}, error) {
+	for i := range svcs {
+		switch svc := svcs[i].(type) {
+		case string:
+			return svc, nil
+		case *did.Service:
+			return svc, nil
+		}
```

Comment:
Can these be combined into [one case](https://github.com/golang/go/wiki/Switch#multiple-cases) statement as these are doing same things ?

Reply:
done